lib/kms/index.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+const AWS = require('aws-sdk');
+const crypto = require('crypto');
+
+/**
+ * Validates the KMS keys provided in the Config.
+ *
+ * @returns {Promise}
+ */
+const check = () => {
+  const regions = Config.get('aws:region');
+  const validatedRegions = Object.keys(regions).map((region) => {
+    const key = regions[region];
+    const KMS = new AWS.KMS({region});
+
+    return new Promise((resolve, reject) => {
+      KMS.describeKey({KeyId: key}, (err, data) => {
+        if (err) {
+          resolve(false);
+        }
+
+        resolve({region, key});
+      });
+    });
+  });
+
+  return Promise.all(validatedRegions).then((regions) => {
+    const filtered = regions.filter((r) => !!r);
+
+    if (filtered.length === 0) {
+      throw new Error('No valid KMS keys');
+    }
+
+    return filtered;
+  });
+};
+
+/**
+ * Validates that retrieving the plaintext datakey and performing a decrypt on your cipher secret
+ * returns the expected original secret
+ * @param params
+ * @returns {Promise}
+ */
+const round_trip_decrypt = (params) => new Promise((resolve, reject) => {
```

Comment:
Why do a round trip on the secret to see if it can be decrypted? I guess my thinking here is that an ACS server doesn't need KMS permission to decrypt a secret, since it's only going to be handling encryption.